db/document.go

```diff
@@ -301,30 +301,30 @@ func (s *syncData) GetSyncCas() uint64 {
 }
 
 // syncData.IsSGWrite - used during feed-based import
-func (s *syncData) IsSGWrite(cas uint64, rawBody []byte) bool {
+func (s *syncData) IsSGWrite(dbStats *DatabaseStats, cas uint64, rawBody []byte) bool {
```

Comment:
I don't like this kind of method signature bloat just for stats tracking - it feels clunky and makes the code less readable everywhere this gets called, as well as breaking some of the functional isolation of this code.  

This seems like a case where if we want to report more granular stats information with db context, we could change the API to return the information we want to log, and have callers that hold a db context do the stats logging.  

i.e. in this case, change the API to IsSGWrite(cas uint64, rawBody []byte) (isSGWrite bool, crc32Match bool) and only maintain a stat for base.StatKeyCrc32cMatchCount.